src/main/java/com/example/account/event/Event.java

```diff
@@ -16,10 +16,17 @@
 @Table("event")
 @NoArgsConstructor
 @AllArgsConstructor
-public abstract class Event {
+public class Event {
```

Comment:
maybe a better design option is to make Event interface, and implement it with specific one

Reply:
interface marker?

Reply:
@amerkul no, I mean something like this:
```java
interface Event {
void on(); // or can be use an Aggregate
}

class UserCreated implements Event {
@Override
void on() {
// logic here
}
}
```

Reply:
Can I use an aggregate, but return an event object? Or it's important to use a void method?
```java
class Event {
// id, type, datetime, json
}

interface Aggregate {
Event process(Command);
}

class UserCreated implements Aggregate {
@Override
Event process(Command) {
// logic here
}
}
```

Reply:
Also good, but in my example, I return `void` because the method "manipulates" the data, simply it's a verb. `process` also can be void. you can also rename this method to `Event event(Command command)`.
By the way it is not good to use concrete classes like `Event.java` in your Aggregate interface